src/hssm/plotting/posterior_predictive.py

```diff
@@ -189,21 +191,98 @@ def _plot_posterior_predictive_2D(
             title="",
             label_order=["observed", "predicted"],
         )
 
     if title:
-        g.fig.subplots_adjust(top=0.9)
-        g.fig.suptitle(title)
+        g.figure.subplots_adjust(top=0.9)
+        g.figure.suptitle(title)
 
     g.set_xlabels(xlabel)
     g.set_ylabels(ylabel)
 
     return g
 
 
+def _process_linestyles_pp(linestyles):
+    if isinstance(linestyles, str):
+        return [linestyles, linestyles]
+
+    if isinstance(linestyles, list) or isinstance(linestyles, tuple):
+        if all(isinstance(ls, str) for ls in linestyles):
+            if len(linestyles) == 1:
+                return [linestyles[0], linestyles[0]]
+            elif len(linestyles) == 2:
+                return linestyles
+            else:
+                raise ValueError(
+                    "The `linestyles` argument must be a string or a list of length 1 "
+                    "or 2."
+                )
+        else:
+            raise ValueError(
+                "The `linestyles` argument must be a string or a list of strings."
+            )
+
+    if isinstance(linestyles, dict):
+        if not set(linestyles.keys()).issubset({"predicted", "observed"}):
+            raise ValueError(
+                "The keys of the `linestyles` dictionary must be 'predicted' and/or "
+                "'observed'."
+            )
+        linestyles = [linestyles.get("predicted", "-"), linestyles.get("observed", "-")]
+    else:
+        raise ValueError(
+            "The `linestyles` argument must be a string, a list of strings,"
+            " or a dictionary."
+        )
+    return linestyles
+
+
+def _process_linewidths_pp(linewidths):
+    if isinstance(linewidths, float):
+        return [linewidths, linewidths]
+
+    if isinstance(linewidths, list) or isinstance(linewidths, tuple):
+        if all(isinstance(lw, float) for lw in linewidths):
+            if len(linewidths) == 1:
+                return [linewidths[0], linewidths[0]]
+            elif len(linewidths) == 2:
+                return linewidths
+            else:
+                raise ValueError(
+                    "The `linewidths` argument must be a float or a list of length 1 or"
+                    " 2."
+                )
+        else:
+            raise ValueError(
+                "The `linewidths` argument must be a float or a list of floats."
+            )
+
+    if isinstance(linewidths, dict):
+        if not set(linewidths.keys()).issubset({"predicted", "observed"}):
+            raise ValueError(
+                "The keys of the `linewidths` dictionary must be 'predicted' and/or"
+                " 'observed'."
+            )
+        linewidths = [
+            linewidths.get("predicted", 1.25),
+            linewidths.get("observed", 1.25),
+        ]
+    else:
+        raise ValueError(
+            "The `linewidths` argument must be a float, a list of floats,"
+            "or a dictionary."
+        )
+    return linewidths
```

Comment:
This function's logic is almost identical as that of `_process_linestyles_pp`'s. Could a more abstract one be defined and called in either of them?